src/api/app/models/workflow/step.rb

```diff
@@ -24,21 +24,6 @@ def target_package
     Package.find_by_project_and_name(target_project_name, target_package_name)
   end
 
-  # FIXME: Remove this and use create_subscriptions and update_subscriptions as soon as BranchPackageStep and Token::Workflow are refactored
-  #        This method is public since it has to be called from Token::Workflow. This shouldn't be needed.
-  def create_or_update_subscriptions(package, workflow_filters)
```

Comment:
This method is still used here:
https://github.com/openSUSE/open-build-service/blob/a334e384ecc45aed5c9a7234a26ec78490c34757/src/api/app/models/token/workflow.rb#L66

Reply:
I wonder if it's really a good idea to split `create_or_update_subscriptions` into two separate methods just to follow the KISS principle. This is forcing to add more conditions with repeated code.

This piece of code

```
if scm_webhook.updated_pull_request?
  if target_package.blank?
    create_target_package
    create_subscriptions(target_package, workflow_filters)
  else
    update_subscriptions(target_package, workflow_filters)
  end
elsif scm_webhook.new_pull_request?
  create_target_package
  create_subscriptions(target_package, workflow_filters)
end
 ```
 
 can be simplified to:

```
create_target_package if scm_webhook.new_pull_request? || (scm_webhook.updated_pull_request? && target_package.blank?)
create_or_update_subscriptions(target_package, workflow_filters)
 ```

@dmarcoux  Would you agree if I bring back `create_or_update_subscriptions` only?

Reply:
I'm fine with having `create_or_update_subscriptions`, as long as we don't keep `create_subscriptions` and `update_subscriptions` around.

Reply:
Ok. Code updated! The commits are different. Please have another look